README.md

```diff
@@ -1,8 +1,161 @@
-[![Build Status](https://travis-ci.org/tgu/kb_uploadmethods.svg?branch=master)](https://travis-ci.org/tgu/kb_uploadmethods)
-
 # kb_uploadmethods
----
 
-Uploaders for kbase datatypes, especially for files in stageing area
+[![Build Status](https://travis-ci.org/tgu/kb_uploadmethods.svg?branch=master)](https://travis-ci.org/tgu/kb_uploadmethods)  
+[Release Notes](RELEASE_NOTES.md)
+
+## Description
+This module implements [KBase](https://www.kbase.us) apps that are used to transform data files into KBase data objects for use in analysis. This also includes a few apps for adding external files to the user's [staging area](https://docs.kbase.us/getting-started/narrative/add-data#uploading-data-from-external-sources), and manipulating them there.
+
+See [here](https://docs.kbase.us/data/upload-download-guide/uploads) for more information on uploading data to KBase.
+
+## Development
+This module was created using the KBase SDK. See the [documentation](https://kbase.github.io/kb_sdk_docs/) for more detail. Apps are mostly broken down into submodules.
+
+The main entrypoint is in `lib/kb_uploadmethods/kb_uploadmethodsImpl.py`. Individual apps in that module use one or more utility module to handle different file types. These are all under `lib/kb_uploadmethods/Utils`. Add a new Util module if you're adding an uploader for a new data type.
+
+## Testing
+This module can be tested with the following steps, common to all KBase SDK modules.
+1. Install the [KBase SDK](https://kbase.github.io/kb_sdk_docs/).
+2. Fetch this module and navigate to it from the console.
+3. Run `kb-sdk test` once - this will initialize the `test_local/test.cfg` file.
+4. Populate the `test_local/test.cfg` file with a KBase authentication token.
```

Comment:
similar to @eamahanna's comment above - it would be helpful to share how to do get an auth token, unless it's already covered in the KB SDK docs

Reply:
Yes, it is covered in the kb-sdk docs. Officially our policy is that only approved developers are allowed to get a token.

Reply:
Added a link to the docs about getting a dev token.